### PR TITLE
Remove device role if role not sent

### PR DESCRIPTION
**Pull request by afairclo**
[mqtt.py](https://github.com/MeshAddicts/meshinfo/blob/43ae523acd59c1c1042d4ed45633c8e4b3e81dcc/mqtt.py#L406) is updating node role info when a role is included with the nodeinfo message, but it appears nodes that are set to Client don't send  "role: 0" in their nodeinfo message.

If device is configured for any other role and is ingested by Meshinfo, and later configured for Client, Meshinfo is not removing the previously set role.

### CODE REVIEW

**Review by kevinelliott — APPROVED** [review]
Thanks for the contribution! This looks good.

**Review by afairclo — COMMENTED** [review]
Missing colon on else statement

**Review by afairclo — COMMENTED** [review]
Fix zIndex placement

File: templates/static/map.html.j2 [hunk 1]
```diff
@@ -88,7 +88,8 @@
           color: 'white',
           width: 2
         })
-      })
+      }),
+	    zIndex: 0
```
Comment (kevinelliott) [inline templates/static/map.html.j2:92]: Fix: tab is wrong for these. Can you adjust?

**Review by kevinelliott — CHANGES_REQUESTED** [review]
Just need to fix the tabs on the indexes.

**Review by afairclo — COMMENTED** [review]
Adjusted

**Review by afairclo — COMMENTED** [review]
Indentation adjustments

**Comment by kevinelliott** [comment]
Looks great! Thanks @afairclo ... merging.

**Pull request merged**